### PR TITLE
NIM-15 Update frontend to pass JSON for player move

### PR DESCRIPTION
**Pull request by dodecaphonix**

### CODE REVIEW

File: nim/templates/index.html [hunk 1]
```diff
@@ -29,27 +29,43 @@ <h1>So it begins...</h1>
...
+    </div>
+
+    <div>
+      <input type="number" id="pile" value="1" min="1">
+      pile
+    <div>
+    </div>
+      <input type="number" id="stones" value="1" min="1">
+      stones
+    </div>
```
Comment (romainrichard) [inline nim/templates/index.html]: Looks like those should be before the `Make Move` button. Also same thing for below, the inputs should be before the `New Game` button.

File: nim/static/game_operations.js [hunk 2]
```diff
@@ -13,18 +13,44 @@ function newGame() {
...
+    updatePage( json.playerState )
+  }
+
+  if ( json.botState && !$('#delay').is(":checked") ) {
+    setTimeout(function() { updatePage( json.botState ) }, 100);
+  }
+
+  else if ( json.botState ) {
+    setTimeout(function() { updatePage( json.botState ) }, 1000);
+  }
```
Comment (romainrichard) [inline nim/static/game_operations.js]: I think you should be able to factorize this. maybe:
```
  if ( json.botState ) {
    setTimeout(function() { updatePage( json.botState ) }, $('#delay').is(":checked")? 1000: 100);
  }
```

File: nim/app.py [hunk 3]
```diff
@@ -21,15 +21,15 @@ def index():
...
+@app.route('/update', methods=['GET', 'POST'])
 def update_game_state():
     state = session['state']
     # player_move = loads('{"pile": 0, "stones": 1}')
     # app.game.update(state, player_move)
     bot_move = app.game.chaos.move(state)
     app.game.update(state, bot_move)
     session['state'] = state
-    return jsonify({'state': state})
+    return jsonify({'playerState': None, 'botState': state})
```
Comment (romainrichard) [inline nim/app.py]: I don't get this. The game only has one state :confused:

File: nim/templates/index.html [hunk 4]
```diff
@@ -29,27 +29,43 @@ <h1>So it begins...</h1>
...
 
     <div>
-      <input type="number" id="maxPerPile" value="12" min="1">
+      <input type="number" id="maxPerPile" value="12" min="1" onchange="newGame()">
       maximum stones per pile
     </div>
 
+    <div>
+      <input type="checkbox" id="delay">
+      delay for bot move
```
Comment (romainrichard) [inline nim/templates/index.html]: Err ok, so it just makes our game slower? Why?

File: nim/app.py (hunk 3 above)
Comment (dodecaphonix) [reply nim/app.py]: Pushing refactor. No longer applicable.

File: nim/static/game_operations.js (hunk 2 above)
Comment (dodecaphonix) [reply nim/static/game_operations.js]: Pushing refactor. No longer applicable.

File: nim/templates/index.html (hunk 4 above)
Comment (dodecaphonix) [reply nim/templates/index.html]: Pushing refactor. No longer applicable.

File: nim/templates/index.html (hunk 1 above)
Comment (dodecaphonix) [reply nim/templates/index.html]: Done! :)

File: nim/static/game_operations.js [hunk 5]
```diff
@@ -1,4 +1,7 @@
-var maxPiles = 5;
+var pileLeft = '<div style="float: left;"><h1>';
+var pileRight = '</h1></div>';
```
Comment (romainrichard) [inline nim/static/game_operations.js]: We only use those once, why have a variable for them?

**Comment by romainrichard** [comment]
@dodecaphonix Maybe it's time to merge this PR?

**Pull request merged**